code/modules/clothing/head/jobs.dm

```diff
@@ -276,52 +299,64 @@
 		balloon_alert(user, "too big!")
 		return
 
-	var/input = tgui_input_text(user, "What is the activation phrase?", "Activation phrase", "gadget", max_length = 26)
-	if(!input)
-		return
-	if(input in items_by_phrase)
-		balloon_alert(user, "already used!")
+	var/desired_phrase = tgui_input_text(user, "What is the activation phrase?", "Activation phrase", "gadget", max_length = 26)
+	if(!desired_phrase)
 		return
 
 	if(item.loc != user || !user.transferItemToLoc(item, src))
 		return
 
-	to_chat(user, span_notice("You install [item] into the [thtotext(contents.len)] slot in [src]."))
+	to_chat(user, span_notice("You install [item] into the [thtotext(contents.len)] slot of [src]."))
 	playsound(src, 'sound/machines/click.ogg', 30, TRUE)
-	items_by_phrase[input] = item
+	set_phrase(desired_phrase,item)
+
+	return TRUE
+
+
 
 /obj/item/clothing/head/fedora/inspector_hat/attack_self(mob/user)
 	. = ..()
-	var/phrase = tgui_input_list(user, "What item do you want to remove by phrase?", "Item Removal", items_by_phrase)
-	if(!phrase)
+	if(!length(items_by_regex))
+		return
+	var/list/found_items = list()
+	for(var/found_regex in items_by_regex)
+		found_items += items_by_regex[found_regex]
+	var/obj/found_item = tgui_input_list(user, "What item do you want to remove?", "Item Removal", found_items)
+	if(!found_item)
 		return
-	user.put_in_inactive_hand(items_by_phrase[phrase])
+	user.put_in_inactive_hand(found_item)
 
 /obj/item/clothing/head/fedora/inspector_hat/AltClick(mob/user)
 	. = ..()
 	var/new_prefix = tgui_input_text(user, "What should be the new prefix?", "Activation prefix", prefix, max_length = 24)
 	if(!new_prefix)
 		return
-	prefix = new_prefix
+	set_prefix(new_prefix)
 
 /obj/item/clothing/head/fedora/inspector_hat/Exited(atom/movable/gone, direction)
 	. = ..()
-	for(var/phrase in items_by_phrase)
-		var/obj/item/result = items_by_phrase[phrase]
-		if(gone == result)
-			items_by_phrase -= phrase
-			return
+	for(var/found_regex in items_by_regex)
+		var/obj/item/found_item = items_by_regex[found_regex]
+		if(gone != found_item)
+			continue
+		items_by_regex -= found_regex
+		phrases_by_regex -= found_regex
+		break
 
 /obj/item/clothing/head/fedora/inspector_hat/atom_destruction(damage_flag)
-	for(var/phrase in items_by_phrase)
-		var/obj/item/result = items_by_phrase[phrase]
-		result.forceMove(drop_location())
-	items_by_phrase = null
-	return ..()
+
+	var/atom/A = drop_location()
+	for(var/found_regex in items_by_regex)
+		var/obj/item/result = items_by_regex[found_regex]
+		result.forceMove(A)
+		items_by_regex -= found_regex
+		phrases_by_regex -= found_regex
+
+	. = ..()
 
 /obj/item/clothing/head/fedora/inspector_hat/Destroy()
-	QDEL_LIST_ASSOC(items_by_phrase)
-	return ..()
+	QDEL_LIST_ASSOC(items_by_regex) //Anything that failed to drop gets deleted.
+	. = ..()
```

Comment:
```suggestion
	return ..()
```

they both literally mean the same thing since BYOND will auto append `return .` to any proc that has the `.` value set so let's just be clear here

Reply:
(actually it will also `return .` if `.` is unset (`null`) but that's another thing entirely, there's just zero reason to change this regardless)